strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/ArtifactManagementService.java

```diff
@@ -222,12 +224,19 @@ private void validateUploadedChecksumAgainstCache(byte[] checksum,
 
         if (!matchesChecksum(checksum, artifactBasePath, checksumExtension))
         {
+            String checksumStr = new String(checksum, StandardCharsets.UTF_8);
+
             logger.error("The checksum for {} [{}] is invalid!",
                          artifactPath,
-                         new String(checksum, StandardCharsets.UTF_8));
-        }
+                         checksumStr);
+
+            if ("Maven 2".equals(repository.getLayout()) && repository.isStrictChecksumValidation())
```

Comment:
Please, use `Maven2LayoutProvider.ALIAS` for this.
Also, is it possible to move this check somewhere within the Maven layout provider's module's code? This service should be generic.

Reply:
I have been searching for a place in Maven layout provider's module to perform "strictChecksumValidation". I found class ArtifactMetadataServiceImpl but not sure whether it is the best option. Any idea or inspiration that you could suggest guys? I am bit lost.
Thanks!

Reply:
@ptirador : Any advice?

Reply:
In my opinion, this could be done like this:
1. Replace this condition by a line of code like this:
```
validateChecksumForLayoutProvider(repository, checksumStr, artifactPath);
```

1.1 This method would be defined like this:
```
protected void validateChecksumForLayoutProvider(Repository repository,
                                                 String checksumStr,
                                                 String artifactPath)
            throws ArtifactStorageException
{
        // Empty for this generic class.
}
```

2. Create a class `org.carlspring.strongbox.services.impl. MavenArtifactManagementServiceImpl` in `strongbox-storage-maven-layout-provider`, which would extends from `ArtifactManagementService`.
2.1. Create a method `validateChecksumForLayoutProvider` in this class, which will override the method in `ArtifactManagementService`, adding this specific condition.

3. Search the uses of `ArtifactManagementService` in Maven-related classes (`MavenArtifactController`, etc) and replace them for the specific component `MavenArtifactManagementServiceImpl`.

Another option would be similar to this, but overriding the whole `validateUploadedChecksumAgainstCache` method instead of creating that method that I mentioned.

Yeah, maybe it's not the best solution (specially because of the last step), but I can't think of another solution right now...

Reply:
> Also, is it possible to move this check somewhere within the Maven layout provider's module's code? This service should be generic.

this is not assumed to be layout specific. If it really should then we need a followup taks for it.

@carlspring can you please provide an example why you think it should depends on layout?

Reply:
> **this** is not assumed to be layout specific. If it really should then we need a follow up tasks, but I'm not sure we need it now.

@sbespalov what do you mean about "this", you mean "strict checksum validation", so "strict checksum validation" should be generic for all layouts?